nodebuilder/p2p/reachability.go

```diff
@@ -0,0 +1,37 @@
+package p2p
+
+import (
+	"context"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/network"
+)
+
+const reachabilityCheckTick = 10 * time.Second
+
+func reachabilityCheck(ctx context.Context, host HostBase) {
+	getter, ok := host.(autoNatGetter)
+	if !ok {
+		panic("host does not implement autoNatGetter")
+	}
+	autoNAT := getter.GetAutoNat()
+
+	go func() {
+		ticker := time.NewTicker(reachabilityCheckTick)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				reachability := autoNAT.Status()
+				if reachability == network.ReachabilityPublic {
+					return
+				}
+
+				log.Error("Host is not reachable from the public network")
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
```

Comment:
Why go routine is needed here?

Reply:
> Every ten second scream in logs that node is not reachable, only if its FN/BN

Reply:
It seems the context provided to the call is construction time context. So it will only do the check while other components are being launched. Which implies that ctx.Done may happen before the first check even performed. 

Would it better to do it in sync mode (without go-routine)? Also would you consider to terminate the node if check failed?

Reply:
> It seems the context provided to the call is construction time context.

This is likely confusing, but its not `Start`-like context. Its a context that is alive for as long as node lives.

Reply:
> Also would you consider to terminate the node if check failed?

This is possible but tricky to do right. It's not the most secure reachability check protocol(AutoNATv1) and can be abused. I am also not confident that the results of the check are always gonna be correct. It also needs sometime to run, you need to connect and send requests to peers(re sync)

I believe the current solution is simple enough and should show allow us to test this properly and after we can consider to block on it if this solution gonna be more reliable(likely with AutoNATv2 which is in development)